src/ILLink.CodeFix/DAMCodeFixProvider.cs

```diff
@@ -90,52 +90,49 @@ public sealed override FixAllProvider GetFixAllProvider ()
 			return WellKnownFixAllProviders.BatchFixer;
 		}
 
-		public override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		public override Task RegisterCodeFixesAsync (CodeFixContext context)
 		{
 			var document = context.Document;
-			if (await document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false) is not { } root)
-				return;
```

Comment:
Operating on a syntax tree is okay to do in `RegisterCodeFixesAsync`. It's only getting a `SemanticModel` object that is recommended against.